LLVMFloatToFixed/IndirectCallConversion.cpp

```diff
@@ -0,0 +1,78 @@
+#include "LLVMFloatToFixedPass.h"
+#include "TypeUtils.h"
+#include "llvm/ADT/SmallPtrSet.h"
+#include "llvm/IR/Constants.h"
+#include "llvm/IR/Function.h"
+#include "llvm/IR/InstIterator.h"
+#include "llvm/IR/Instructions.h"
+#include "llvm/IR/Intrinsics.h"
+#include "llvm/IR/Module.h"
+#include "llvm/Support/raw_ostream.h"
+#include <llvm/Transforms/Utils/Cloning.h>
+#include <llvm/Transforms/Utils/ValueMapper.h>
+
+using namespace llvm;
+using namespace taffo;
+using namespace flttofix;
+
+/// Retrieve the indirect calls converted into trampolines and re-use the
+/// original indirect functions.
+void FloatToFixed::convertIndirectCalls(llvm::Module &m) {
+  std::vector<llvm::CallInst *> trampolineCalls;
+
+  // Retrieve the trampoline calls using the INDIRECT_METADATA
+  for (llvm::Function &curFunction : m) {
+    for (auto instructionIt = inst_begin(curFunction);
+         instructionIt != inst_end(curFunction); instructionIt++) {
+      if (auto curCallInstruction =
+              llvm::dyn_cast<llvm::CallInst>(&(*instructionIt))) {
+        if (curCallInstruction->getMetadata(INDIRECT_METADATA)) {
+          trampolineCalls.push_back(curCallInstruction);
+        }
+      }
+    }
+  }
+
+  // Convert the trampoline calls
+  for (auto trampolineCall : trampolineCalls) {
+    auto *metadata = dyn_cast<ValueAsMetadata>(
+        *trampolineCall->getMetadata(INDIRECT_METADATA)->op_begin());
+    auto *indirectFunction = dyn_cast<Function>(metadata->getValue());
+
+    if (indirectFunction->getName() == "__kmpc_fork_call") {
```

Comment:
This check is different from the one applied in Initializer. Can we uniform them?

Reply:
I added a map similar to the one in Initializer. The syntax is a bit different because we are using pointers to member functions of the FloatToFixed struct here, but still it is more consistent across the two passes now.